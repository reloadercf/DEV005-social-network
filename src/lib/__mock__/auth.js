/* eslint-disable prefer-promise-reject-errors */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */

export const addUserToSocialNetwork = jest.fn(() => Promise.resolve({ message: 'success', email: 'ok' }));
