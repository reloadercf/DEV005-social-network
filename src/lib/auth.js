/* eslint-disable prefer-promise-reject-errors */
import { createUserWithEmailAndPassword } from 'firebase/auth';
import { auth } from './firebaseConfig.js';

export const addUserToSocialNetwork = (email, password) => new Promise((resolve, reject) => {
  createUserWithEmailAndPassword(auth, email, password)
    .then((userCredential) => {
    // Signed in
      const user = userCredential.user;
      resolve({ message: 'success', email: user.email });
    // ...
    })
    .catch((error) => {
      const errorMessage = error.message;
      reject({ message: 'error', email, errorMessage });
    });
});
