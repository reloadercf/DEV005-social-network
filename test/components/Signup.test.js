/**
 * @jest-environment jsdom
 */

import signup from '../../src/components/signUp.js';
import * as auth from '../../src/lib/auth.js';

describe('signUp', () => {
  test('snapshot of signup', () => {
    const DOM = document.createElement('div');
    DOM.append(signup());
    expect(DOM).toMatchSnapshot();
  });
  test('is a function', () => {
    expect(typeof signup).toBe('function');
  });
  test('have a button', () => {
    const DOM = document.createElement('div');
    DOM.append(signup());
    const haveAButton = DOM.querySelector('#return');
    expect(haveAButton).not.toBe(undefined);
  });
  test('after click button return call function navigateTo', () => {
    const DOM = document.createElement('div');
    const navigateTo = jest.fn();
    DOM.append(signup(navigateTo));
    const buttonBack = DOM.querySelector('#return');
    buttonBack.click();
    expect(navigateTo).toHaveBeenCalledTimes(1);
  });
  test('after click button return call function navigateTo with / ', () => {
    const DOM = document.createElement('div');
    const navigateTo = jest.fn();
    DOM.append(signup(navigateTo));
    const buttonBack = DOM.querySelector('#return');
    buttonBack.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/');
  });
});

describe('Button SAVE', () => {
  test('Test of click button save', (done) => {
    jest.spyOn(auth, 'addUserToSocialNetwork').mockImplementation(() => Promise.resolve({ message: 'success', email: 'carlos@carlos.com' }));
    const DOM = document.createElement('div');
    DOM.append(signup());
    const email = DOM.querySelector('#email');
    const password = DOM.querySelector('#password');
    const answer = DOM.querySelector('#answer');
    email.value = 'carlos@carlos.com';
    password.value = '123456';

    const buttonSave = DOM.querySelector('#save');
    buttonSave.click();
    expect(auth.addUserToSocialNetwork).toHaveBeenCalledTimes(1);
    expect(auth.addUserToSocialNetwork).toHaveBeenLastCalledWith('carlos@carlos.com', '123456');
    setTimeout(() => {
      expect(answer.classList.contains('success')).toBe(true);
      expect(answer.classList.contains('error')).not.toBe(true);
      done();
    }, 0);
  });
});
