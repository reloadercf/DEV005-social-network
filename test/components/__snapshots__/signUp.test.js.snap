// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp snapshot of signup 1`] = `
<div>
  <div>
    
      
    <div
      class="login"
      id="signup"
    >
      
        
      <span>
        Signup
      </span>
      
        
      <input
        id="email"
        name="email"
        placeholder="email"
        type="email"
      />
      
        
      <input
        id="password"
        name="password"
        placeholder="***"
        type="password"
      />
      
        
      <span
        id="answer"
      />
      
        
      <button
        id="save"
      >
        Save
      </button>
      
      
    </div>
    
      
    <button
      id="return"
    >
      Return
    </button>
    
    
  </div>
</div>
`;
